### PR TITLE
[FLINK-34952][cdc-composer][sink] Flink CDC pipeline supports SinkFunction

### PR DESCRIPTION
**Pull request by loserwang1024**
### What's the problem
Though current Flink CDC pipeline define com.ververica.cdc.common.sink.FlinkSinkFunctionProvider to to provide a Flink SinkFunction for writing events to external systems. However, com.ververica.cdc.runtime.operators.sink.DataSinkWriterOperator don't support SouceFunction, which means sink implement SinkFunction cannot use CDC pipeline.
Why not support  SourceFunction in Flink CDC pipeline ?

### How to solve
Add An operator `DataSinkOperator` that processes records to be written into a {@link org.apache.flink.streaming.api.functions.sink.SinkFunction}.

### CODE REVIEW

**Comment by loserwang1024** [comment]
@leonardBang , @LYanquan , @PatrickRen , CC

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/org/apache/flink/cdc/connectors/values/sink/ValuesDataSinkOptions.java [hunk 1]
```diff
@@ -35,4 +35,10 @@ public class ValuesDataSinkOptions {
                     .booleanType()
                     .defaultValue(true)
                     .withDescription("True if the Event should be print to console.");
+
+    public static final ConfigOption<Boolean> LEGACY_ENABLED =
+            ConfigOptions.key("legacy.enabled")
```
Comment (PatrickRen) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/org/apache/flink/cdc/connectors/values/sink/ValuesDataSinkOptions.java:40]: The word `legacy` is too vague. Sink V1 could also be a legacy one. What about using an enumeration here listing all APIs that the value sink supports?

File: flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/sink/DataSinkOperator.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+ * An operator that processes records to be written into a {@link
+ * org.apache.flink.streaming.api.functions.sink.SinkFunction}.
+ *
+ * <p>The operator is a proxy of {@link org.apache.flink.streaming.api.operators.StreamSink} in
+ * Flink.
+ *
+ * <p>The operator is always part of a sink pipeline and is the first operator.
+ */
+@Internal
+public class DataSinkOperator extends StreamSink<Event> {
```
Comment (PatrickRen) [inline flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/sink/DataSinkOperator.java:51]: What about `DataSinkFunctionOperator`?

**Review by PatrickRen — COMMENTED** [review]
@loserwang1024 Thanks for the contribution! I left some comments.

File: flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/sink/DataSinkOperator.java (hunk 2 above)
Comment (loserwang1024) [reply flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/sink/DataSinkOperator.java]: Done it.

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/org/apache/flink/cdc/connectors/values/sink/ValuesDataSinkOptions.java (hunk 1 above)
Comment (loserwang1024) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/org/apache/flink/cdc/connectors/values/sink/ValuesDataSinkOptions.java]: Add `SinkType`

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/org/apache/flink/cdc/connectors/values/sink/ValuesDataSinkOptions.java [hunk 3]
```diff
@@ -35,4 +35,10 @@ public class ValuesDataSinkOptions {
                     .booleanType()
                     .defaultValue(true)
                     .withDescription("True if the Event should be print to console.");
+
+    public static final ConfigOption<String> SINK_TYPE =
+            ConfigOptions.key("sinkType")
```
Comment (PatrickRen) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/org/apache/flink/cdc/connectors/values/sink/ValuesDataSinkOptions.java:40]: What about `sink.api`? We use the word "type" in YAML for specifying the category of connector, such as "mysql", "doris", "values", so I'm concerned about the potential conflict.

**Review by PatrickRen — COMMENTED** [review]
@loserwang1024 Thanks for the update! Just a minor comment

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/org/apache/flink/cdc/connectors/values/sink/ValuesDataSinkOptions.java (hunk 3 above)
Comment (loserwang1024) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-values/src/main/java/org/apache/flink/cdc/connectors/values/sink/ValuesDataSinkOptions.java]: It sounds better. Just do it.

**Review by PatrickRen — APPROVED** [review]
@loserwang1024 Thanks for the update! LGTM

**Pull request merged**